dislib/classification/rf/decision_tree.py

```diff
@@ -160,52 +128,45 @@ def get_groups(sample, y_s, features_mmap, index, value):
     return left, y_l, right, y_r
 
 
-def build_leaf(y_s, tree_path):
-    frequencies = Counter(y_s)
-    most_common = frequencies.most_common(1)
-    if most_common:
-        mode = most_common[0][0]
-    else:
-        mode = None
-    return Leaf(tree_path, len(y_s), frequencies, mode)
+def compute_leaf_info(y_s, n_classes):
+    frequencies = np.bincount(y_s, minlength=n_classes)
+    mode = np.argmax(frequencies)
+    return LeafInfo(len(y_s), frequencies, mode)
 
 
-def split_node(tree_path, sample, n_features, y_s, n_classes, m_try,
+def split_node(sample, n_features, y_s, n_classes, m_try,
                samples_file=None, features_file=None):
     if features_file is not None:
-        return split_node_a(tree_path, sample, n_features, features_file, y_s,
-                            n_classes, m_try)
+        return split_node_a(sample, n_features, y_s, n_classes, m_try,
+                            features_file)
     elif samples_file is not None:
-        return split_node_b(tree_path, sample, n_features, samples_file, y_s,
-                            n_classes, m_try)
+        return split_node_b(sample, n_features, y_s, n_classes, m_try,
+                            samples_file)
     else:
-        raise ValueError('Invalid combination of arguments. samples_file is'
-                         ' None and features_file is None.')
+        raise ValueError('Invalid combination of arguments. samples_file is '
+                         'None and features_file is None.')
 
 
-@task(features_file=FILE_IN, returns=(InternalNode, list, list, list, list))
-def split_node_a(tree_path, sample, n_features, features_file, y_s, n_classes,
-                 m_try):
+@task(features_file=FILE_IN, returns=(object, list, list, list, list))
+def split_node_a(sample, n_features, y_s, n_classes, m_try, features_file):
```

Comment:
What is the difference between split_node_a and split_node_b? Better give a more explanatory name

Reply:
PyCOMPSs doesn't allow for optional FILE parameters. This is circumvented by adding multiple tasks. In this case, one task becomes 2 tasks and more functions. I've renamed the tasks to  _split_node, _split_node_with_features and the containing function _split_node_wrapper.